### PR TITLE
Notification2.0

### LINKED ISSUE #22: Notification on best by date

### PR DESCRIPTION
**Pull request by AlexTerhar**
i edit the suggested changes from Felix. 
Due to the Android oreo Version, is it not possible to add a green cicle behind the logo, because the background will remain transparent.
The default chanel is set in the manifest. 

Closes #22

### CODE REVIEW

File: app/src/main/java/app/wefridge/wefridge/MainActivity.kt [hunk 1]
```diff
@@ -39,6 +45,15 @@ class MainActivity : AppCompatActivity() {
...
         val user = FirebaseAuth.getInstance().currentUser ?: return authWall()
 
+        //to add the user to a personal Topic
+        Firebase.messaging.subscribeToTopic(user.uid)
+            .addOnCompleteListener { task ->
+                Log.d("Main_Activity: ", if (task.isSuccessful)
+                    "Subscribed to ${user.uid}" else "Failed to subscribe")
+            }
+
+
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MainActivity.kt:55]: Remove blank lines

File: app/src/main/java/app/wefridge/wefridge/WeFridgeFirebaseMessagingService.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        super.onMessageReceived(remoteMessage)
+
+        if (remoteMessage.notification !=null) {
+            Log.d("FCMService", "Message data payload: ${remoteMessage.data}")
+            generateNotification(remoteMessage.notification!!.title!!,remoteMessage.notification!!.body!!)
+        }
+    }
+
+    /**
+     * Called the FCM registration token at the beginning of the installation.
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/WeFridgeFirebaseMessagingService.kt:39]: I don't understand this comment. Does it mean "called **by** the FCM registration token", or "**Call** the FCM registration" or something else

File: app/src/main/res/layout/notification.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            android:src="@drawable/show_fridge_notification" />
+
+        <TextView
+            android:id="@+id/title_logo"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_toRightOf="@id/app_logo"
+            android:layout_marginLeft="10dp"
+            android:layout_marginTop="15dp"
+            android:text="Title"
```
Comment (frcroth) [inline app/src/main/res/layout/notification.xml:25]: can be removed I think

File: app/src/main/res/layout/notification.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            android:textSize="20sp"
+            android:textStyle="bold"/>
+        <TextView
+            android:id="@+id/message"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_below="@id/title_logo"
+            android:layout_toRightOf="@id/app_logo"
+            android:layout_marginLeft="10dp"
+            android:text="the text description"
```
Comment (frcroth) [inline app/src/main/res/layout/notification.xml:35]: Can be removed I think

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -11,9 +11,9 @@
     <string name="Pantry">Pantry</string>
     <string name="Nearby">Nearby</string>
     <string name="Settings">Settings</string>
-    <string name="EditItem">Edit Item</string>
     <string name="hello_second_fragment">Hello second fragment. Arg: %1$s</string>
-
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (frcroth) [inline app/src/main/res/values/strings.xml:16]: Remove

**Review by frcroth — COMMENTED** [review]
You don't need to create a new PR, just push further changes to the branch and it will update and all subscribed will get notifications

File: app/src/main/res/layout/notification.xml (hunk 4 above)
Comment (AlexTerhar) [reply app/src/main/res/layout/notification.xml:35]: no, the whole notification script is needed, there is the connection in getRemoteView. To create the notification

**Comment by frcroth** [comment]
I mean the random text

File: app/src/main/java/app/wefridge/wefridge/MainActivity.kt [hunk 6]
```diff
@@ -1,5 +1,6 @@
 package app.wefridge.wefridge
 
+import android.annotation.SuppressLint
```
Comment (frcroth) [inline app/src/main/java/app/wefridge/wefridge/MainActivity.kt:3]: Remove

**Review by frcroth — APPROVED** [review]

**Review by henrijaeger — APPROVED** [review]

**Review by PassiDel — APPROVED** [review]
I applied the following things in 344d1d4f:

- Un/Subscribe to user/owner topic
- Refactor MessagingService
- Create MessagingChannel on App launch (translated name)
- Remove unused Strings

**Comment by PassiDel** [comment]
Oh, right, you're on the road right now. I'll take over the merging.
@AlexTerhar

**Pull request merged**